src/logger.ts

```diff
@@ -1,16 +1,39 @@
 import { APIOrder } from '@0x/types';
 import * as pino from 'pino';
 
-import { LOG_LEVEL, LOGGER_INCLUDE_TIMESTAMP, MAX_ORDER_EXPIRATION_BUFFER_SECONDS } from './config';
+import {
+    LOG_ASYNC_ENABLED,
+    LOG_ASYNC_FLUSH_INTERVAL_SEC,
+    LOG_BUFFER_BYTES,
+    LOG_LEVEL,
+    LOGGER_INCLUDE_TIMESTAMP,
+    MAX_ORDER_EXPIRATION_BUFFER_SECONDS,
+} from './config';
 import { ONE_SECOND_MS } from './constants';
 import { ExpiredOrderError } from './errors';
 
-export const logger = pino({
-    level: LOG_LEVEL,
-    useLevelLabels: true,
-    timestamp: LOGGER_INCLUDE_TIMESTAMP,
+const pinoDestination = pino.destination({
+    minLength: LOG_BUFFER_BYTES, // Buffer before writing
+    sync: !LOG_ASYNC_ENABLED, // Asynchronous logging
 });
 
+export const logger = pino(
+    {
+        level: LOG_LEVEL,
+        useLevelLabels: true,
+        timestamp: LOGGER_INCLUDE_TIMESTAMP,
+    },
+    pinoDestination,
+);
+
+if (LOG_ASYNC_ENABLED) {
+    // asynchronously flush every 10 seconds to keep the buffer empty
+    // in periods of low activity
+    setInterval(() => {
+        logger.flush();
+    }, ONE_SECOND_MS * LOG_ASYNC_FLUSH_INTERVAL_SEC).unref();
```

Comment:
What is "unref" exactly?

Reply:
When called the `Timeout` object will not require the Node.js event loop to remain active. If there is no other activity keeping the event loop running, the process may exit before the Timeout object's callback is invoked. Which means this will not prevent nodejs process from exiting before callback is invoked.

[0] - https://nodejs.org/api/timers.html#timers_timeout_unref

Reply:
cool!